graylog2-web-interface/src/components/common/Drawer.tsx

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { Drawer as MantineDrawer } from '@mantine/core';
+import styled, { css } from 'styled-components';
+
+const StyledDrawer = styled(MantineDrawer)(({ theme }) => css`
+  .mantine-Drawer-content, .mantine-Drawer-header {
+    background-color: ${theme.colors.global.contentBackground};
+  }
+`);
+type Props = Pick<React.ComponentProps<typeof MantineDrawer>, 'opened' | 'onClose' | 'position' | 'size' | 'children' | 'title' | 'closeOnClickOutside'>;
+const Drawer = ({ title, ...props }: Props) => <StyledDrawer offset={15} padding="lg" radius={5} zIndex={1032} title={title} {...props} />;
```

Comment:
If I remember it correctly we talked about the preference to not use the rest operator for props and having an explicit component interface, because it for example benefits having just a few variations of the component in the product.

Reply:
The props which are usable are limited by the `Props` type above it, so it is not simply passing through _all_ possible props.